sympy/core/quaternion.py

```diff
@@ -340,11 +355,12 @@ def pow_cos_sin(self, p):
         q2 = Quaternion.from_axis_angle(v, p * angle)
         return q2 * (q.norm() ** p)
 
-    def diff(self, *args):
-        return Quaternion(diff(self.a, *args), diff(self.b, *args),
-                          diff(self.c, *args), diff(self.d, *args))
+    def _eval_derivative(self, x):
```

Comment:
I had already added _eval_Integral an _eval_diff on line 161. I let those methods call diff and integrate.

Reply:
Sorry I didn't notice that